spring-batch-docs/asciidoc/whatsnew.adoc

```diff
@@ -4,69 +4,150 @@
-* `StaxEventItemWriter` - `StaxEventItemWriterBuilder`
-* `SynchronizedItemStreamReader` - `SynchronizedItemStreamReaderBuilder`
+== What's New in Spring Batch 4.1
+
+The Spring Batch 4.1 release adds the following features:
+
+* A new `@SpringBatchTest` annotation to simplify testing batch components
+* A new `@EnableBatchIntegration` annotation to simplify remote chunking configuration
+* A new `JsonItemReader` to support JSON format
+
+[[whatsNewTesting]]
+=== `@SpringBatchTest` annotation
+
+Spring Batch provides some nice utility classes (such as the `JobLauncherTestUtils` and
+`JobRepositoryTestUtils`) and test execution listeners (`StepScopeTestExecutionListener`
+and `JobScopeTestExecutionListener`) to test batch components. However, in order
+to use these utilities, you must configure them explicitly. This release introduces
+a new annotation named `@SpringBatchTest` that automatically adds utility beans and
+listeners to the test context and makes them  available for autowiring,
+as the following example shows:
+
+[source, java]
+----
+@RunWith(SpringRunner.class)
+@SpringBatchTest
+@ContextConfiguration(classes = {JobConfiguration.class})
+public class JobTest {
+
+   @Autowired
+   private JobLauncherTestUtils jobLauncherTestUtils;
+
+   @Autowired
+   private JobRepositoryTestUtils jobRepositoryTestUtils;
+
+
+   @Before
+   public void clearMetadata() {
+      jobRepositoryTestUtils.removeJobExecutions();
+   }
+
+   @Test
+   public void testJob() throws Exception {
+      // given
+      JobParameters jobParameters =
+            jobLauncherTestUtils.getUniqueJobParameters();
+
+      // when
+      JobExecution jobExecution =
+            jobLauncherTestUtils.launchJob(jobParameters);
+
+      // then
+      Assert.assertEquals(ExitStatus.COMPLETED,
+                          jobExecution.getExitStatus());
+   }
+
+}
+----
+
+For more details about this new annotation, see the
+<<testing.adoc#creatingUnitTestClass,Unit Testing>> chapter.
+
+[[whatsNewIntegration]]
+=== `@EnableBatchIntegration` annotation
+
+Setting up a remote chunking job requires the definition of a number of beans:
+
+* A connection factory to acquire connections from the messaging middleware (JMS, AMQP, and others)
+* A `MessagingTemplate` to send requests from the master to the workers and back again
+* An input channel and an output channel for Spring Integration to get messages from the messaging middleware
+* A special item writer (`ChunkMessageChannelItemWriter`) on the master side that knows how to send chunks of data to workers for processing and writing
+* A message listener (`ChunkProcessorChunkHandler`) on the worker side to receive data from the master
+
+This can be a bit daunting at first glance. This release introduces a new annotation
+named `@EnableBatchIntegration` as well as new APIs (`RemoteChunkingMasterStepBuilder`
+ and `RemoteChunkingWorkerBuilder`) to simplify the configuration. Here is an example:
```

Comment:
Change "Here is an example:" to "The following example shows how to use the new annotation and APIs:"

(The idea behind having "following" in all of these leader sentences is to draw the reader along to the next chunk. It's a minor point, but now we need it for consistency. Thanks for doing it.)